Develop

Nexler0 opened the pull request:
Sprint 12

Feat:
add-reviews
add-search
add-recommendations
add-director
add-most-populars
add-remove-endpoint
add-common-films

Commit:
Sprint 12

Change H2 driver to v2.1.210
Delete all constrain in DB

Commit:
Sprint 12

Change H2 driver to v2.1.210
Delete all constrain in DB

EvgenyGH committed:
feat: add endpoint GET /users/{id}/recommendations. Add getRecommendations method into FilmService and FilmDbStorage.

EvgenyGH committed:
feat: add endpoint GET /users/{id}/recommendations. Add getRecommendations method into FilmService and FilmDbStorage.

BykovIvan committed:
feat: реализация списка для директоров для фильмов

BykovIvan committed:
feat: добавлние в фильмсторадж реализации

BykovIvan committed:
feat: делаю реализацию GetAllFilms с директорами

BykovIvan committed:
feat: сделана работа с директорами, но getFIlm выдает только 2 фильма

BykovIvan committed:
feat: начало работы с получением фильмов директора по лайкам и рейтигу

EvgenyGH committed:
feat: add endpoint GET /users/{id}/recommendations test.

BykovIvan committed:
feat: реализация методов получения фильмов

BykovIvan committed:
feat: пора деkать тесты

Commit:
Sprint 12

Feat : create add-reviews

BykovIvan committed:
feat: начала добавления тестов

BykovIvan committed:
feat: начала добавления тестов

EvgenyGH committed:
feat: add endpoint GET /users/{id}/recommendations test.

EvgenyGH committed:
Merge remote-tracking branch 'origin/develop' into develop

# Conflicts:
#	db/filmorate.mv.db
#	db/filmorate.trace.db
#	pom.xml

BykovIvan committed:
feat: добавлены тесты контроллеру директорам

BykovIvan committed:
feat: доработки и исправления

BykovIvan committed:
feat: доработки и исправления

Commit:
Sprint 12

Feat : create add-reviews

Nexler0 committed:
Удаление db директории

Commit:
Sprint 12

Feat : create add-reviews

Commit:
Merge remote-tracking branch 'origin/add-reviews' into add-reviews

Nexler0 committed:
Delete db directory

Nexler0 committed:
Merge pull request #10 from Nexler0/add-reviews

Add-reviews

BykovIvan committed:
feat: правка в gitignore

BykovIvan committed:
feat: check gitignore

BykovIvan committed:
feat: check gitignore

BykovIvan committed:
feat: delete db

BykovIvan committed:
Merge branch 'add-director' into develop

# Conflicts:
#	src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java
#	src/main/java/ru/yandex/practicum/filmorate/storage/FilmStorage.java
#	src/main/java/ru/yandex/practicum/filmorate/storage/dao/FilmDbStorage.java
#	src/main/resources/schema.sql

Commit:
Sprint 12

Fix : fix bug with likes and get films

EvgenyGH committed:
feat: add endpoint GET /films/common?userId={userId}&friendId={friendId} with test.

Commit:
add search_film_filter

Commit:
add search_film_filter

EvgenyGH committed:
Merge pull request #13 from Nexler0/add-common-films

Добавил функциональность «Общие фильмы»

Commit:
change search_method title

Commit:
Sprint 12

Fix : resolve tests problem in legacy code

Commit:
Merge remote-tracking branch 'origin/develop' into develop

Commit:
fix DBStorage method

Commit:
add order-sort-for-search-result

Commit:
merge add-search

BykovIvan committed:
feat: валится на 2 тестах из за того что не правильно формируется жанры

BykovIvan committed:
feat: постман тесты проходят

BykovIvan committed:
Merge pull request #18 from Nexler0/add-most-populars

Add most populars by Year and Genre

Commit:
Sprint 12

First review

Feat:
add-reviews
add-search
add-recommendations
add-director
add-most-populars
add-remove-endpoint
add-common-films

Commit:
Sprint 12

First review

Feat:
add-reviews
add-search
add-recommendations
add-director
add-most-populars
add-remove-endpoint
add-common-films

Fix : getRate return likesId.size()

Commit:
Sprint 12

First review

Feat:
add-reviews
add-search
add-recommendations
add-director
add-most-populars
add-remove-endpoint
add-common-films

Fix : delete field rate in film and refactor

Commit:
Sprint 12

First review

Feat:
add-reviews
add-search
add-recommendations
add-director
add-most-populars
add-remove-endpoint
add-common-films

Fix :
- delete user and his id in another users friend list
- get last not used users_id
- delete not used fields

kukolev commented on src/main/java/ru/yandex/practicum/filmorate/controller/DirectorController.java:
```diff
@@ -0,0 +1,116 @@
+     * Запрос на создание нового директора, добавление его в таблицу
+     * Post add new director and add him to DB
+     * @param director
+     * @return
```
Здесь нужно либо дописать информацию о возвращаемом значении, либо убрать строчку @return. Сейчас этот код не имеет смысла и только отвлекает внимание.

Аналогично нужно поправить везде ниже.

kukolev commented on src/main/java/ru/yandex/practicum/filmorate/storage/DirectorDao.java:
```diff
@@ -0,0 +1,64 @@
+     * Add directors in to DB
+     *
+     * @param director
+     * @return
```
Здесь нужно либо дописать информацию о возвращаемом значении, либо убрать строчку @return. Сейчас этот код не имеет смысла и только отвлекает внимание.

Аналогично нужно поправить везде ниже.

kukolev commented on src/main/java/ru/yandex/practicum/filmorate/storage/FilmDirectorsDao.java:
```diff
@@ -0,0 +1,57 @@
+     *
+     * @param filmId
+     * @param directorId
+     * @return
```
Здесь нужно либо дописать информацию о возвращаемом значении, либо убрать строчку @return. Сейчас этот код не имеет смысла и только отвлекает внимание.

Аналогично нужно поправить везде ниже.

kukolev commented on src/main/java/ru/yandex/practicum/filmorate/storage/FilmStorage.java:
```diff
@@ -21,4 +24,31 @@ public interface FilmStorage {
+     *
+     * @param directorId
+     * @param param
+     * @return
```
Здесь нужно либо дописать информацию о возвращаемом значении, либо убрать строчку @return. Сейчас этот код не имеет смысла и только отвлекает внимание.

Аналогично нужно поправить везде ниже.

kukolev commented on src/main/java/ru/yandex/practicum/filmorate/storage/dao/FilmDirectorsDaoImpl.java:
```diff
@@ -0,0 +1,83 @@
+
+    @Override
+    public void addDirectorToFilm(Integer filmId, Integer directorId) {
+        String insertQuery = "insert into FILMS_DIRECTORS (FILM_ID, DIRECTOR_ID) values (?, ?)";
```
Эти запросы лучше оформить как константы, тогда строка с запросом не будет создаваться каждый раз при вызове метода.

Аналогично нужно поправить ниже.

kukolev commented on src/main/java/ru/yandex/practicum/filmorate/controller/ErrorHandler.java:
```diff
@@ -1,33 +1,42 @@
+
+    @ExceptionHandler({GetRecommendedFilmsErrorException.class})
+    @ResponseStatus(HttpStatus.INTERNAL_SERVER_ERROR)
+    public ErrorResponse GetRecommendedFilmsErrorExceptionHandler(final GetRecommendedFilmsErrorException exception) {
```
Нарушение кодстайла: название методов должно быть выполнено в стиле camelCase.

kukolev commented on src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java:
```diff
@@ -46,4 +48,66 @@ public String deleteTheMovieLike(Integer id, Integer userId) {
+    }
+
+    public List<Film> getPopularFilmsByGenre(Integer count, Integer genreId) {
+        if (genreId < 1 || genreId > 6) {
```
Числа 1 и 6 лучше сделать константами, чтобы по названию можно было понять за что они отвечают.

Commit:
Sprint 12

First review

Feat:
add-reviews
add-search
add-recommendations
add-director
add-most-populars
add-remove-endpoint
add-common-films

Accept reviewer comments

Commit:
Sprint 12

First review

Feat:
add-reviews
add-search
add-recommendations
add-director
add-most-populars
add-remove-endpoint
add-common-films

Accept reviewer comments
